examples/property_prediction/pretrain_gnns/chem/utils.py

```diff
@@ -0,0 +1,72 @@
+import torch
+import dgl
+import numpy as np
+
+
+def find_neighbor_edges(g, node_id):
+    """Given a node with its graph, return all the edges connected to that node."""
+    predecessors = g.predecessors(node_id)
+    successors = g.successors(node_id)
+    predecessors_edges = g.edge_ids(predecessors, torch.full(predecessors.shape, node_id, dtype=torch.long))
+    successors_edges = g.edge_ids(successors, torch.full(successors.shape, node_id, dtype=torch.long))
+    return torch.cat((predecessors_edges, successors_edges))
+
+
+def mask_edges(g, masked_nodes_indices):
+    """Given a graph and masked nodes, mask all edges that connected to the masked nodes and return these edge indices."""
+    masked_edges_indices = []
+    for masked_nodes_index in masked_nodes_indices:
+        masked_edges_indices.extend(find_neighbor_edges(g, masked_nodes_index))
+    return torch.LongTensor(masked_edges_indices)
+
+
+class PretrainMaskingMoleculeCSVDataset(object):
```

Comment:
I guess we can merge `PretrainMaskingMoleculeCSVDataset` and `PretrainSupervisedMoleculeCSVDataset`?

Reply:
done